lightning/src/chain/keysinterface.rs

```diff
@@ -807,8 +820,10 @@ impl BaseSign for InMemorySigner {
 		Ok((sign(secp_ctx, &msghash, &self.node_secret), sign(secp_ctx, &msghash, &self.funding_key)))
 	}
 
-	fn ready_channel(&mut self, channel_parameters: &ChannelTransactionParameters) {
-		assert!(self.channel_parameters.is_none(), "Acceptance already noted");
+	fn provide_channel_parameters(&mut self, channel_parameters: &ChannelTransactionParameters) {
+		if self.channel_parameters.is_some() {
```

Comment:
@TheBlueMatt see previous discussion with @wpaulino in this PR about this.  the PR as it currently stands calls `provide_channel_parameters` twice in `funding_signed` and `funding_provided`, so the doc change you suggested conflicts.  these extra calls are needed for `InMemorySigner`, because the instances are distinct, but for VLS it would be considered a duplicate call because we dispatch to a central signer instance.

I think we can still make this work by defining the API here to say that multiple calls are OK if the the calls are idempotent (same params provided).

If we go in this direction, please derive `PartialEq` for `ChannelTransactionParameters`, so that it's easy to check.

Reply:
I added a comment https://github.com/lightningdevkit/rust-lightning/pull/1867/files#r1040014478 where I think the docs may be a bit confusing.

Reply:
looks good